seacrowd/sea_datasets/uit_visfd/uit_visfd.py

```diff
@@ -0,0 +1,237 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Crawled textual feedback from customers about smartphones on \
+    a large e-commerce website in Vietnam. The label of the \
+    dataset is ten aspects and three polarities. Please read \
+    the guidelines in the paper for more information. \
+    We randomly divide the dataset into three sets:
+"""
+import os
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+import pandas as pd
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Tasks, Licenses
+
+# TODO: Add BibTeX citation
+_CITATION = """\
+  @InProceedings{10.1007/978-3-030-82147-0_53,
+  author="Luc Phan, Luong
+  and Huynh Pham, Phuc
+  and Thi-Thanh Nguyen, Kim
+  and Khai Huynh, Sieu
+  and Thi Nguyen, Tham
+  and Thanh Nguyen, Luan
+  and Van Huynh, Tin
+  and Van Nguyen, Kiet",
+  editor="Qiu, Han
+  and Zhang, Cheng
+  and Fei, Zongming
+  and Qiu, Meikang
+  and Kung, Sun-Yuan",
+  title="SA2SL: From Aspect-Based Sentiment Analysis to Social Listening System for Business Intelligence",
+  booktitle="Knowledge Science, Engineering and Management ",
+  year="2021",
+  publisher="Springer International Publishing",
+  address="Cham",
+  pages="647--658",
+  abstract="In this paper, we present a process of building a social listening system based on aspect-based sentiment analysis in Vietnamese, from creating a dataset to building a real application. Firstly, we create UIT-ViSFD, a Vietnamese Smartphone Feedback Dataset, as a new benchmark dataset built based on a strict annotation scheme for evaluating aspect-based sentiment analysis, consisting of 11,122 human-annotated comments for mobile e-commerce, which is freely available for research purposes. We also present a proposed approach based on the Bi-LSTM architecture with the fastText word embeddings for the Vietnamese aspect-based sentiment task. Our experiments show that our approach achieves the best performances (in F1-score) of 84.48{\%} for the aspect task and 63.06{\%} for the sentiment task, which performs several conventional machine learning and deep learning systems. Lastly, we build SA2SL, a social listening system based on the best performance model on our dataset, which will inspire more social listening systems in the future.",
+  isbn="978-3-030-82147-0"
+  }
+"""
+
+_DATASETNAME = "uit_visfd"
+
+_DESCRIPTION = """
+UIT-ViSFD is the Vietnamese Smartphone Feedback Dataset.
+It is an aspect-based sentiment analysis dataset.
+It consists of 11,122 human-annotated comments for mobile e-commerce.
+"""
+
+_HOMEPAGE = "https://github.com/LuongPhan/UIT-ViSFD"
+
+_LANGUAGES = ["vie"]
+
+_LICENSE = Licenses.UNKNOWN.value
+
+_LOCAL = False
+
+_URLS = {_DATASETNAME: "https://github.com/LuongPhan/UIT-ViSFD/raw/main/UIT-ViSFD.zip"}
+
+_SUPPORTED_TASKS = [Tasks.SENTIMENT_ANALYSIS]
```

Comment:
```
======================================================================
ERROR: runTest (__main__.TestDataLoader)
Run all tests that check:
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/frederikus-h/Library/CloudStorage/GoogleDrive-fredyhudi@gmail.com/My Drive/_self_research/2023.12 seacrowd/seacrowd-datahub/tests/test_seacrowd.py", line 145, in setUp
    self.datasets_seacrowd[schema] = datasets.load_dataset(
  File "/Users/frederikus-h/anaconda3/envs/seacrowd/lib/python3.10/site-packages/datasets/load.py", line 2556, in load_dataset
    builder_instance = load_dataset_builder(
  File "/Users/frederikus-h/anaconda3/envs/seacrowd/lib/python3.10/site-packages/datasets/load.py", line 2265, in load_dataset_builder
    builder_instance: DatasetBuilder = builder_cls(
  File "/Users/frederikus-h/anaconda3/envs/seacrowd/lib/python3.10/site-packages/datasets/builder.py", line 371, in __init__
    self.config, self.config_id = self._create_builder_config(
  File "/Users/frederikus-h/anaconda3/envs/seacrowd/lib/python3.10/site-packages/datasets/builder.py", line 592, in _create_builder_config
    raise ValueError(
ValueError: BuilderConfig 'uit_visfd_seacrowd_text' not found. Available: ['uit_visfd_source', 'uit_visfd_seacrowd_text_multi']

----------------------------------------------------------------------
Ran 1 test in 1.171s

FAILED (errors=1)
```
Please follow the [Dataloader implementation guidelines](https://github.com/SEACrowd/seacrowd-datahub/blob/master/DATALOADER.md), especially from [section 4 onwards](https://github.com/SEACrowd/seacrowd-datahub/blob/master/DATALOADER.md#4-check-if-your-dataloader-works).

Reply:
I've already checked and tested it before with additional parameters while testing with `--schema TEXT_MULTI`. 

From now on, I change the `SEACROWD_SCHEMA_NAME` from `text` to `text_multi`, due to the `Tasks` section on https://github.com/SEACrowd/seacrowd-datahub/issues/627 that says **Aspect Based Sentiment Analysis**.

Solved on https://github.com/SEACrowd/seacrowd-datahub/pull/637/commits/315c5db9424778495a546e21686f8297ace01b50.

Thanks for the review, Sir!

Reply:
> I change the SEACROWD_SCHEMA_NAME from text to text_multi

It does not seem the `SEACROWD_SCHEMA_NAME` was changed, but rather was introduced from [the commit](https://github.com/SEACrowd/seacrowd-datahub/commit/315c5db9424778495a546e21686f8297ace01b50#diff-2be3ac8c0d5d422375cb00b1e7de9b5af7355235458e0e96f3912dfbe6ed63e3R98).

Also, the SEACrowd schema was already correct, i.e.: text_multi.

Reply:
Yup, instead of hard-coded on the each SEACrowd schema, I decided to create constant `SEACROWD_SCHEMA_NAME` to ease the changes.

I change Tasks from `SENTIMENT_ANALYSIS` to `ASPECT_SENTIMENT_ANALYSIS` to incorporate the SEACrowd schema name, from `text` to `text_multi`.

Reply:
> I change Tasks from SENTIMENT_ANALYSIS to ASPECT_SENTIMENT_ANALYSIS to incorporate the SEACrowd schema name, from text to text_multi.

Do note that this dataset is aspect based sentiment analysis.

Reply:
Sorry for the typo, what I meant is: `ASPECT_BASED_SENTIMENT_ANALYSIS` instead of `ASPECT_SENTIMENT_ANALYSIS`.